tests/system/Database/ConfigTest.php

```diff
@@ -0,0 +1,183 @@
+<?php
+namespace CodeIgniter\Database;
+
+use CodeIgniter\Database\BaseConnection;
+use CodeIgniter\Database\Config;
+use CodeIgniter\Test\ReflectionHelper;
+
+class DatabaseConfig extends \CodeIgniter\Test\CIUnitTestCase
+{
+	use ReflectionHelper;
+
+	protected $group = [
+		'DSN'      => '',
+		'hostname' => 'localhost',
+		'username' => 'first',
+		'password' => 'last',
+		'database' => 'dbname',
+		'DBDriver' => 'MySQLi',
+		'DBPrefix' => 'test_',
+		'pConnect' => true,
+		'DBDebug'  => (ENVIRONMENT !== 'production'),
+		'cacheOn'  => false,
+		'cacheDir' => 'my/cacheDir',
+		'charset'  => 'utf8',
+		'DBCollat' => 'utf8_general_ci',
+		'swapPre'  => '',
+		'encrypt'  => false,
+		'compress' => false,
+		'strictOn' => true,
+		'failover' => [],
+	];
+
+	protected $dsnGroup = [
+		'DSN'      => 'MySQLi://user:pass@localhost/dbname?DBPrefix=test_&pConnect=true&charset=latin1&DBCollat=latin1_swedish_ci',
+		'hostname' => '',
+		'username' => '',
+		'password' => '',
+		'database' => '',
+		'DBDriver' => 'SQLite3',
+		'DBPrefix' => 't_',
+		'pConnect' => false,
+		'DBDebug'  => (ENVIRONMENT !== 'production'),
+		'cacheOn'  => false,
+		'cacheDir' => 'my/cacheDir',
+		'charset'  => 'utf8',
+		'DBCollat' => 'utf8_general_ci',
+		'swapPre'  => '',
+		'encrypt'  => false,
+		'compress' => false,
+		'strictOn' => true,
+		'failover' => [],
+	];
+
+	protected $dsnGroupPostgre = [
+		'DSN'      => 'Postgre://user:pass@localhost/dbname?DBPrefix=test_&connect_timeout=5&sslmode=1',
```

Comment:
should have with port example ?:

```suggestion
              'DSN'      => 'Postgre://user:pass@localhost:5432/dbname?DBPrefix=test_&connect_timeout=5&sslmode=1',
```